hoodie-client/src/main/java/com/uber/hoodie/table/HoodieMergeOnReadTable.java

```diff
@@ -94,4 +114,99 @@ public HoodieMergeOnReadTable(HoodieWriteConfig config,
         }
     }
 
+    @Override
+    public List<HoodieRollbackStat> rollback(JavaSparkContext jsc, List<String> commits) throws IOException {
+
+        Map<String, HoodieInstant> commitsAndCompactions =
+                this.getActiveTimeline().getInstants()
+                        .filter(i -> Sets.newHashSet(HoodieActiveTimeline.COMMIT_ACTION, HoodieActiveTimeline.COMPACTION_ACTION, HoodieActiveTimeline.DELTA_COMMIT_ACTION).contains(i.getAction()))
+                        .filter(i -> commits.contains(i.getTimestamp())).collect(Collectors.toMap(i -> i.getTimestamp(), i -> i));
+
+        // Atomically un-publish all non-inflight commits
+        commitsAndCompactions.entrySet().stream().map(entry -> entry.getValue())
+                .filter(i -> !i.isInflight()).forEach(this.getActiveTimeline()::revertToInflight);
+
+        logger.info("Unpublished " + commits);
+
+        List<HoodieRollbackStat> allRollbackStats = jsc.parallelize
+                (FSUtils.getAllPartitionPaths(FSUtils.getFs(), this.getMetaClient().getBasePath(), config.shouldAssumeDatePartitioning()))
```

Comment:
if we have already logged the affected partitions in either commit metadata or the workload profile information in inflight files, why list the partitions again?

Reply:
In general, we can move to using commitMetadata for both cleaning and rollbacks I think. We did not do this initially, since we wanted to gain some confidence and just relied on the backing fs, to provide these

Reply:
So I did not change it to read from inflight files since I kept the parallelism at a partition level and then use commit information to delete/append rollback blocks after this parallelization. This would have to be changed to loop over all commits -> read commit metadata -> parallelize on partitions read from commit metadata, this might lead to a different sequence of executions for rollback from what it is today (given sparkcontext.parallelize(..) is invoked multiple times).

Reply:
Understand there is going to be code changes needed here.. but seems better to switch to consuming the commit metadata still.